src/opentime/timeRange.h

```diff
@@ -82,6 +82,34 @@ class TimeRange {
         return _start_time < other.end_time_exclusive() && other._start_time < end_time_exclusive();
     }
 
+    bool before(TimeRange other) const {
+        return _start_time < other._start_time && end_time_exclusive() <= other._start_time;
+    }
+
+    bool before(RationalTime other) const {
+        return end_time_exclusive() < other;
+    }
+
+    bool meets(TimeRange other) const {
+        return end_time_exclusive() == other._start_time || _start_time == other.end_time_exclusive();
+    }
+
+    bool begins(TimeRange other) const {
+        return _start_time == other._start_time;
```

Comment:
This is another general comment: floating point equality comparisons are problematic at best. to be genuinely useful, these routines should all have a tolerance value, with a default tolerance value of double 0.f to force exact equivalence. A question for all is whether tolerance is straight up tolerance, or if there needs to be some sort of clamping behavior in addition to the tolerance. In other words, do we need something like

```
bool begins(TimeRange other, RationalTime tolerance);
```
where tolerance.value is the tolerance value, tolerance.rate is the rate at which to test that tolerance. A typical value might be RationalTime(1./48., 1/24) - half a frame tolerance, with an expected rate of 24 fps. 

Any thoughts on this?

Reply:
Problems generally occur due to rounding errors during arithmetic operations, right? Would it make sense to put in effort and handle it like they do in geometry intersection routines like here: http://www.pbr-book.org/3ed-2018/Shapes/Managing_Rounding_Error.html or is half a frame of tolerance better? And if we're usinf half a frame of tolerance, would we still need to pass in the tolerance value in the function?
Couldn't we just use `RationalTime(0.5/rate(), 1.0/rate())` directly?

Reply:
In my response below, I've added an epsilon parameter to the predicates to address this. Thanks!